### ISSUE TITLE
Thoughts about release note URLs

### ISSUE THREAD

**Issue by thomas-mc-work** [mag37/dockcheck#103]
I've just tested the latest release regarding the release notes feature. For me it didn't work as expected. My email body looked like this:

    lego-web-1  ->  url missing
    nextcloud-cron-1  ->  url missing
    nextcloud-db-1  ->  url missing
    www-web-1  ->  url missing

That wasn't very satisfying :-) Wouldn't it be better to match the release note URL by the image URL? Conatiner names can vary, but the image source URL is canonical.

Another thing I've noticed is that the release notes URLs aren't printed when running the check via the console. Is there a reaoson not to do that?

Despite that I think this could become a great and very useful feature! Thank you.

**Comment by mag37** [comment]
Thank you for testing and good points.

The `urls.list` was created as a template/example first but then it seemed like a good idea to just continue to append to it.

I agree, image names might have been more "standard" even though they also vary - many different sources of the same service. So they'd need one entry for each source. Could still work though, so good point.

And another reason is that the list of containers with updates are just container names currently, not referencing image so it was easier to just iterate through that.
Could still work if I rewrote it to add all names + images to an array/table of some sort. Though it's pretty cumbersome already to do the lookups/correlation.

**Comment by mag37** [comment]
Please look at this discussion for further testing on the [`notify-rewrite branch`](https://github.com/mag37/dockcheck/discussions/101)

**Comment by thomas-mc-work** [comment]
> Lastly - they're not printed in the main output because I considered it being tied to the notifications. But you're right, maybe there should be an option for it!

Maybe the output could be unified?

**Comment by mag37** [comment]
> Maybe the output could be unified?

Yes it could, but I know some people have used the output for custom things - so I rather not clutter that.
Unifying is still a good option - as only people who add the `urls.list` to the script directory will trigger it. And then maybe have a flag to not print the urls to console for those who dont want it (keeping an extra list or just printing column1 isnt much work).

Good points and will consider for future changes! Thank you.

**Comment by thomas-mc-work** [comment]
> Unifying is still a good option - as only people who add the urls.list to the script directory will trigger it.

I've just now realized that this is a reqiurement in order to make it work 😅 So yeah, that condition would be good. Either that, or you reference the source file on github and fetch it every time the script is invoked. But those aspects don't touch the issue with on what to use as the lookup key.
